core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1656,29 +1665,81 @@ public String getName() {
 
         @Override
         public void _run() throws IOException, InstallationStatus {
-            super._run();
+            if (wasInstalled()) {
+                // Do this first so we can avoid duplicate downloads, too
+                // check to see if the plugin is already installed at the same version and skip it
+                LOGGER.warning("Skipping duplicate install of: " + plugin.getDisplayName() + "@" + plugin.version);
```

Comment:
Warning seems a bit much here, as this appears to be a fairly common case. INFO, or probably even lower, seems more appropriate.